### PR TITLE
Add oadm migrate volumesource

### LINKED ISSUE #14677: oadm migrate tool to prevent upgrade if volumeSource.metadata is present
https://github.com/openshift/origin/blob/master/vendor/k8s.io/kubernetes/pkg/api/v1/types.go#L338-L341 has been deprecated for multiple releases.  We should block upgrade if someone has these while we still have the types in 3.6, so that we'll be clean in 3.7.

@liggitt

### PR DESCRIPTION
**Pull request by enj**
`oadm migrate volumesource` confirms that all objects that contain `volumeSource.metadata` do not have it set.  `volumeSource.metadata` is deprecated and will no longer be available in the next release.  Thus all objects that contain a non-nil `volumeSource.metadata` must be manually converted to prevent data loss.

The following resource types are checked by this command:

* cronjobs.batch
* daemonsets.extensions
* deploymentconfigs.apps.openshift.io
* deployments.extensions
* jobs.batch
* podpresets.settings.k8s.io
* pods
* podtemplates
* replicasets.extensions
* replicationcontrollers
* statefulsets.apps

No resources are mutated.

Signed-off-by: Monis Khan <mkhan@redhat.com>

Fixes #14677 

cc @deads2k @smarterclayton @liggitt @simo5 @sdodson @openshift/security 

[test]

### CODE REVIEW

**Comment by enj** [comment]
re[test] flake #14823

File: pkg/cmd/admin/migrate/migrator.go [hunk 1]
```diff
@@ -49,6 +49,8 @@ func AlwaysRequiresMigration(_ *resource.Info) (Reporter, error) {
 	return ReporterBool(true), nil
 }
 
+var NotChanged = ReporterBool(false)
```
Comment (deads2k) [inline pkg/cmd/admin/migrate/migrator.go]: godoc

File: pkg/cmd/admin/migrate/volumesource/volumesource.go [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+		The following resource types are checked by this command:
+
+		* cronjobs
+		* daemonsets
+		* deploymentconfigs
+		* deployments
+		* jobs
+		* podpresets
+		* pods
+		* podsecuritypolicyreviews
```
Comment (deads2k) [inline pkg/cmd/admin/migrate/volumesource/volumesource.go]: not persisted, so it doesn't need to be checked.

File: pkg/cmd/admin/migrate/volumesource/volumesource.go [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+		* podsecuritypolicyselfsubjectreviews
```
Comment (deads2k) [inline pkg/cmd/admin/migrate/volumesource/volumesource.go]: not persisted, so it doesn't need to be checked.

File: pkg/cmd/admin/migrate/volumesource/volumesource.go [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+		* podsecuritypolicysubjectreviews
```
Comment (deads2k) [inline pkg/cmd/admin/migrate/volumesource/volumesource.go]: not persisted, so it doesn't need to be checked.

File: pkg/cmd/admin/migrate/volumesource/volumesource.go [hunk 5]
```diff
@@ -0,0 +1,181 @@
...
+		* podtemplates
+		* replicasets
```
Comment (deads2k) [inline pkg/cmd/admin/migrate/volumesource/volumesource.go]: list groups too.  `replicasets.extensions`

File: pkg/cmd/admin/migrate/volumesource/volumesource.go [hunk 6]
```diff
@@ -0,0 +1,181 @@
...
+}
+
+func NewCmdMigrateVolumeSource(name, fullName string, f *clientcmd.Factory, in io.Reader, out, errout io.Writer) *cobra.Command {
+	options := &MigrateVolumeSourceOptions{
+		ResourceOptions: migrate.ResourceOptions{
+			In:            in,
+			Out:           out,
+			ErrOut:        errout,
+			AllNamespaces: true,
+			Include: []string{
```
Comment (deads2k) [inline pkg/cmd/admin/migrate/volumesource/volumesource.go]: Why is this thing a string and not a `GroupResource`?

**Comment by deads2k** [comment]
Minor comments, lgtm otherwise.  Open a pull that deletes this too and we'll merge it after we branch 3.6.

**Comment by enj** [comment]
[merge][severity:blocker] all comments addressed.

@deads2k can you do a sanity check on the `resource.group` list.

**Comment by enj** [comment]
Opened #14873 for 3.7

**Pull request merged**